varnish/src/vsc_wrapper.rs

```diff
@@ -0,0 +1,67 @@
+use std::ffi::CString;
+use std::mem::size_of;
+use std::ops::{Deref, DerefMut};
+use varnish_sys::ffi::{vsc_seg, VRT_VSC_Alloc, VRT_VSC_Destroy};
+
+pub unsafe trait VscMetric {
+    fn get_metadata() -> &'static str;
+}
+
+pub struct Vsc<T: VscMetric> {
+    metric: *mut T,
+    seg: *mut vsc_seg,
+    name: CString,
+}
+
+impl<T: VscMetric> Vsc<T> {
+    pub fn new(module_name: &str, module_prefix: &str) -> Self {
+        let mut seg = std::ptr::null_mut();
+        let name = CString::new(module_name).expect("module_name contained interior nul byte");
+        let format =
+            CString::new(module_prefix).expect("module_prefix contained interior nul byte");
+
+        let metadata_json = T::get_metadata();
+
+        let metric = unsafe {
+            VRT_VSC_Alloc(
+                std::ptr::null_mut(),
+                &mut seg,
+                name.as_ptr(),
+                size_of::<T>(),
+                metadata_json.as_ptr(),
+                metadata_json.len(),
+                format.as_ptr(),
+                std::ptr::null_mut(),
```

Comment:
```suggestion
                varnish_sys::ffi::va_list::default(),
```

if `NULL` things explode on my laptop